be/src/exec/vectorized/json_scanner.cpp

```diff
@@ -463,13 +463,15 @@ void JsonReader::_reorder_column(std::vector<SlotDescriptor*>* slot_descs,
             continue;
         }
 
-        auto itr = std::find_if(slot_descs->begin(), slot_descs->end(),
-                                [key](const SlotDescriptor* desc) { return desc->col_name() == key; });
+        key_set.insert(key);
 
-        if (itr != slot_descs->end()) {
+        auto itr = std::find_if(ordered_slot_descs.begin(), ordered_slot_descs.end(),
+                                [&key](const SlotDescriptor* desc) { return desc->col_name() == key; });
+
+        if (itr != ordered_slot_descs.end()) {
```

Comment:
Are you trying to swap the element `*iter` with `ordered_slot_descs[idx]`? If so, using `std::swap(*iter, ordered_slot_descs[idx])`

Reply:
Updated.